coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -32,14 +33,20 @@ def __init__(self, documentation, docstyle_definition,
         :param marker:
             The three-element tuple with marker strings, that identified this
             documentation comment.
-        :param range:
-            The position range of type ``TextRange``.
+        :param position:
+            The starting ``TextPosition`` of the documentation.
         """
         self.documentation = documentation
         self.docstyle_definition = docstyle_definition
         self.indent = indent
         self.marker = marker
-        self.range = range
+        self.position = position
+        self.assemble = None if position is None else self._assemble()
```

Comment:
1. As you have an `lru_cache` you can keep `self.assemble()` like it is without renaming it to `self._assemble()`.
2. Also you don't need the position to assemble a docstring.
3. What if I change e.g. markers now? Then we need to update `self.assemble()` again. Or do we consider those fields to be "immutable"?